internal/wasmldr/src/loader/requested.rs

```diff
@@ -0,0 +1,157 @@
+        let pki = PrivateKeyInfo::from_der(&self.0.prvkey)?;
+
+        // Create a relative distinguished name.
+        let rdns = RdnSequence::encode_from_string("CN=localhost")?;
+
+        // Create the extensions.
+        let ku = KeyUsage(KeyUsages::DigitalSignature | KeyUsages::KeyEncipherment).to_vec()?;
+        let eu = ExtendedKeyUsage(vec![ID_KP_SERVER_AUTH, ID_KP_CLIENT_AUTH]).to_vec()?;
+        let bc = BasicConstraints {
+            ca: false,
+            path_len_constraint: None,
+        }
+        .to_vec()?;
+
+        // Create the certificate body.
+        let tbs = TbsCertificate {
+            version: x509::Version::V3,
+            serial_number: UIntBytes::new(&[0u8])?,
+            signature: pki.signs_with()?,
+            issuer: RdnSequence::from_der(&rdns)?,
+            validity: Validity::from_now(Duration::from_secs(60 * 60 * 24 * 365))?,
+            subject: RdnSequence::from_der(&rdns)?,
+            subject_public_key_info: pki.public_key()?,
+            issuer_unique_id: None,
+            subject_unique_id: None,
+            extensions: Some(vec![
+                x509::ext::Extension {
+                    extn_id: ID_CE_KEY_USAGE,
+                    critical: true,
+                    extn_value: &ku,
+                },
+                x509::ext::Extension {
+                    extn_id: ID_CE_BASIC_CONSTRAINTS,
+                    critical: true,
+                    extn_value: &bc,
+                },
+                x509::ext::Extension {
+                    extn_id: ID_CE_EXT_KEY_USAGE,
+                    critical: false,
+                    extn_value: &eu,
+                },
+            ]),
+        };
+
+        // Self-sign the certificate.
+        let alg = tbs.signature;
+        let sig = pki.sign(&tbs.to_vec()?, alg)?;
+        let crt = Certificate {
+            tbs_certificate: tbs,
+            signature_algorithm: alg,
+            signature: BitString::from_bytes(&sig)?,
+        };
+
+        Ok(vec![crt.to_vec()?])
+    }
+
+    pub fn next(self) -> Result<Loader<Attested>> {
+        // If the user supplied
+        let certs = match self.0.config.steward.as_ref() {
+            Some(url) => self.steward(url)?,
+            None => self.selfsigned()?,
+        }
+        .into_iter()
+        .map(rustls::Certificate)
+        .collect::<Vec<_>>();
+
+        // TODO: load this policy from `Config`.
+        let protocol_versions = &[&TLS13];
+        let kx_groups = &[&X25519, &SECP384R1, &SECP256R1];
+        let cipher_suites = &[
+            TLS13_AES_256_GCM_SHA384,
+            TLS13_AES_128_GCM_SHA256,
+            TLS13_CHACHA20_POLY1305_SHA256,
+        ];
+
+        // Set up the server config.
+        let srvcfg = ServerConfig::builder()
+            .with_cipher_suites(cipher_suites)
+            .with_kx_groups(kx_groups)
+            .with_protocol_versions(protocol_versions)?
+            .with_no_client_auth() // TODO: Enable client auth
```

Comment:
We should file an issue about this and add a reference here

Reply:
Done.